flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -22,7 +22,7 @@ constructFlinkClassPath() {
     local FLINK_CLASSPATH
 
     while read -d '' -r jarfile ; do
-        if [[ "$jarfile" =~ .*flink-dist.*.jar ]]; then
+        if [[ "$jarfile" =~ .*/flink-dist[^/]*.jar$ ]]; then
```

Comment:
Did you run into issues with the old regex?

Reply:
I did not run into problems with class loading, but the resulting classpath was different from what is expected by this method.

If I understand this method correctly, it tries to put the dist jar at the end of the class path and the `FLINK_DIST` variable should contain the `flink-dist.jar` not the whole classpath. 

Because I run it within flink source directory within the `flink-dist` module, all jars from the `lib` directory ended up in the `FLINK_DIST` variable, because the old regex matches:
..../**flink-dist**/target/flink-1.11-SNAPSHOT-bin/flink-1.11-SNAPSHOT/lib/*.**jar**

The new regex matches only within the last part of the path.
..../flink-dist/target/flink-1.11-SNAPSHOT-bin/flink-1.11-SNAPSHOT/lib/log4j-1.2.17.jar <- does not match
..../flink-dist/target/flink-1.11-SNAPSHOT-bin/flink-1.11-SNAPSHOT/lib/**flink-dist_2.11-1.11-SNAPSHOT.jar** <- matches

Reply:
ok